### DISCUSSION TITLE
Why does BepInEx need HTTP?

### DISCUSSION THREAD

**Discussion by sosnik** [BepInEx/BepInEx#553] [Help and Q&A]
In the compiled releases for BepInEx there is a `winhttp.dll` file.

Why does BepInEx need HTTP code if it is just injecting into the game?

**Comment by Windows10CE** [comment]
That file is *how* it injects into the game. It contains no HTTP code and is instead https://github.com/NeighTools/UnityDoorstop . Windows searches for this file because Unity uses it, it loads it (which loads doorstop, which loads BepInEx). After Doorstop loads, it forwards everything to the actual winhttp.dll in your system directory again.

**Comment by sosnik** [comment]
Neat, thanks!
